service/frontend/admin_handler.go

```diff
@@ -1860,6 +1860,79 @@ func (adh *AdminHandler) MergeDLQTasks(ctx context.Context, request *adminservic
 	}, nil
 }
 
+func (adh *AdminHandler) DescribeDLQJob(ctx context.Context, request *adminservice.DescribeDLQJobRequest) (*adminservice.DescribeDLQJobResponse, error) {
+	jt := adminservice.DLQJobToken{}
+	err := jt.Unmarshal([]byte(request.JobId))
+	if err != nil {
+		return nil, errInvalidDLQJobToken
+	}
+	client := adh.sdkClientFactory.GetSystemClient()
+	execution, err := client.DescribeWorkflowExecution(ctx, jt.WorkflowId, jt.RunId)
+	if err != nil {
+		return nil, err
+	}
+	response, err := client.QueryWorkflow(ctx, jt.WorkflowId, jt.RunId, dlq.QueryTypeProgress)
+	if err != nil {
+		return nil, err
+	}
+	var queryResponse dlq.ProgressQueryResponse
+	if err = response.Get(&queryResponse); err != nil {
+		return nil, err
+	}
+	var state enumsspb.DLQOperationState
+	switch execution.WorkflowExecutionInfo.Status {
+	case enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING:
+		state = enumsspb.DLQ_OPERATION_STATE_RUNNING
+	case enumspb.WORKFLOW_EXECUTION_STATUS_COMPLETED:
+		state = enumsspb.DLQ_OPERATION_STATE_COMPLETED
+	default:
+		state = enumsspb.DLQ_OPERATION_STATE_FAILED
+	}
+	var opType enumsspb.DLQOperationType
+	switch queryResponse.WorkflowType {
+	case dlq.WorkflowTypeDelete:
+		opType = enumsspb.DLQ_OPERATION_TYPE_PURGE
+	case dlq.WorkflowTypeMerge:
+		opType = enumsspb.DLQ_OPERATION_TYPE_MERGE
+	default:
+		return nil, serviceerror.NewInternal("Invalid DLQ workflow type")
```

Comment:
Please include the type in the error message

Reply:
Done